src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -168,6 +169,10 @@ public function load(array $configs, ContainerBuilder $container): void
             ->addTag('api_platform.uri_variables.transformer');
         $container->registerForAutoconfiguration(ParameterProviderInterface::class)
             ->addTag('api_platform.parameter_provider');
+        $container->registerAttributeForAutoconfiguration(ApiResource::class, static function(ChildDefinition $definition): void {
+            $definition->addTag('api_platform.resource');
+            $definition->addTag('container.excluded', ['source' => __FILE__]);
```

Comment:
Since ApiResource classes are not services but value objects, the tag `container.excluded` is also added, so they cannot be used in autowiring.

Reply:
"source" is rather the file where the attribute is defined

Reply:
(we could add it automatically BTW, PR welcome :)

Reply:
Instead of indicating the source file (which is easy to deduce from the class), we can explain that it is the use of this attribute that causes the class to be excluded.

```suggestion
            $definition->addTag('container.excluded', ['source' => 'with #[ApiResource] attribute']);
```

For context, this is used in [`AutowirePass`](https://github.com/symfony/symfony/blob/6f4f04b98f6134996a57d21d1851d558a34dc45c/src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php#L601).